test/e2e/upgrade_management_cluster.go

```diff
@@ -0,0 +1,138 @@
+package e2e
```

Comment:
Also, this file has to be renamed to `upgrade_management_cluster_test.go` to avoid **undeclared_name** warnings that we recently fixed in https://github.com/metal3-io/cluster-api-provider-metal3/commit/9cb57bd2b19f928032a51ff990fa5ada679de882